pages/react-canvas-components.mdx

```diff
@@ -0,0 +1,256 @@
+  date: 'January 12, 2019',
+};
+
+The `<canvas/>` HTML element can be used to draw graphics with a finer control than the usual DOM or SVG. But with React, trying to draw on a canvas is not intuitive as their interfaces are quite different. With React, each component own their node contrary to canvas where there is only one shared node, and we draw in it. Let's see how we can make a canvas visualization with React components !
+
+## 🧑‍🏫 Canvas 101
+
+The `canvas` element is like a sheet of paper. To draw in real life, you would take a pen, move your hand to a first position, and draw a line by moving your hand to another position. The browser API to draw on a canvas is actually very similar. We first need to make a blueprint of the shape we want to draw (like using a wood pencil), and then we can color it.
+
+```js
+// moving our hand to the starting position
+canvasContext.moveTo(x1, y1);
+// drawing a blueprint line to the finishing position
+canvasContext.lineTo(x2, y2);
+// taking a purple pen and coloring the line
+canvasContext.strokeStyle = 'purple';
+canvasContext.stroke();
+```
+
+Having imperative code like this in a component oriented codebase can be tricky ! Let's create a component that renders a `<canvas />` on the page. Then, we need to call the `moveTo` and `lineTo` methods on it to draw a line. In practice, it's a bit more complicated to bridge those two. We need to use a [React reference](https://thibaut.io/react-refs-evolution) to access the canvas DOM node, then need to get a 2D context out of it, and finally we can call our drawing methods. The code looks like that:
+
+```jsx basic-canvas-hs3cy src/App.js
+const Canvas = () => {
+  // we use a ref to access the canvas' DOM node
+  const canvasRef = React.useRef(null);
+
+  React.useEffect(() => {
+    const context = canvasRef.current.getContext('2d');
+    // ...drawing using the context
+  }, [canvasRef]);
+
+  return <canvas ref={canvasRef} />;
+};
+```
+
+But if we want to draw something a bit more complex, the `Canvas` component can become quite large. Usually, big components are split into several child components. Yet here, this is not possible as there is only one `canvas` node.
+
+## 🎨 Hexagons
+
+To show how to make child components with canvas, let's draw something more fancy ✨
+
+<div
+  style={{
+    height: '360px',
+    width: '100%',
+  }}
+>
+  <RandomHexagons isAnimating={true} />
+</div>
+
+A single hexagon is defined with the folowing data:
+
+- A position on the screen, defined by two `x` and `y` number values.
+- A `radius` that defines its size.
+- A `rotation` so that all hexagons don't look aligned.
+- A `color`.
+
+We need a function that is able to generate some random hexagons to display. The randomisation code is not relevant here, let's assume we have a way to get an array of hexagons. As for how to draw the shape of an hexagon, we need to draw a line between all its corners, and then fill it with a color:
```

Comment:
- Don't need "to display"
- "The randomisation code is not relevant here, let's assume  .." => "The randomisation code is not relevant here; let's just assume  ..."
- " As for how to draw the shape of an hexagon, we need to draw" => " As for how to draw the shape of an hexagon - we need to draw a line between all its corners and then fill it with a color" (ie. add a dash and remove a comma)